atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/TestResourceManager.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.junit.rules.ExternalResource;
+
+import com.google.common.base.Suppliers;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.transaction.api.TransactionManager;
+
+public class TestResourceManager extends ExternalResource implements KvsManager, TmManager {
+    private final Supplier<KeyValueService> getKvsSupplier;
+    private final List<AutoCloseable> closeableResources = new ArrayList<>();
+
+    private KeyValueService keyValueService = null;
+    private TransactionManager transactionManager = null;
+
+    public TestResourceManager(Supplier<KeyValueService> kvsSupplier) {
+        this.getKvsSupplier = Suppliers.memoize(() -> {
+            KeyValueService kvs = kvsSupplier.get();
+            registerCloseable(kvs);
+            return kvs;
+        });
+    }
+
+    public static TestResourceManager inMemory() {
+        return new TestResourceManager(() -> new InMemoryKeyValueService(false));
+    }
+
+    @Override
+    public KeyValueService getDefaultKvs() {
+        return getKvsSupplier.get();
+    }
+
+    @Override
+    public void registerKvs(KeyValueService kvs) {
+        keyValueService = kvs;
+        registerCloseable(kvs);
+    }
+
+    @Override
+    public Optional<KeyValueService> getLastRegisteredKvs() {
+        return Optional.ofNullable(keyValueService);
+    }
+
+    @Override
+    public void registerTm(TransactionManager manager) {
```

Comment:
`registerTm` -> `registerTransactionManager`, and any instance of tm in a method name to `transactionManager`.

 I would even go as far as to say `registerKvs` -> `registerKeyValueService` but I think that's quite ingrained..